File: 10226-HardwoodSpecies.java
//IMPORTANTE!!!  UVA da Runtime Error pero todos los casos de prueba funcionan correctamente, así que lo doy por bueno :)

import java.io.*;
import java.util.*;
public class Main{
    public static void main(String[] args) throws NumberFormatException, IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int cases =  Integer.parseInt(br.readLine());
        br.readLine();
        while(cases-- > 0) {
            //System.out.println("Casos: " + cases);
            int counter = 0;
            Map<String, Integer> dictionary  = new HashMap<String, Integer>();
            String tree = br.readLine();

            try{
                while(!tree.equals("")) {
                counter++;
                if (dictionary.get(tree) == null){
                    
                    dictionary.put(tree, 1);
                } else {
                    dictionary.put(tree, dictionary.get(tree) + 1);
                }

                
                tree = br.readLine();
                

                }
            } catch(NullPointerException e){
                
            }

            List<String> lista = new ArrayList<String>(dictionary.keySet());
            Collections.sort(lista);
            for (String k: lista){
		System.out.printf(k + " %.4f%n",  (dictionary.get(k) * 1.0 / counter) * 100);
            }
        
            if(cases != 0) System.out.println();
        
        }

    
    
    }
}

File: 10226-HardwoodSpecies.py
#IMPORTANTE!!!!  todos los casos de prueba funcionan pero UVA da time limit exceeded :'(
cases = int(input())
input() #The  first blank line

while cases > 0:
    #print("cases: ", cases)
    
    counter = 0
    tree = input() #the name of the tree
    dictionary = {} #we will store the tree and how many times it appears here
    while(tree != "exit() and tree): #while there is no blank line
        #print("Tree name: ", tree)
        counter += 1
        if dictionary.get(tree) == None:
            dictionary[tree] = 1
            #print("valor añadido al diccionatio")
        else:
            dictionary[tree] += 1
            #print("valor modificado al diccionatio")
        try:
            tree = input()
        except:
            break
    
    keys = list(dictionary)
    keys.sort()
    #print("contador: ", counter)
    for k in keys:
        print (k, "{0:.4f}".format((dictionary.get(k) / counter) * 100, 4))
    cases -= 1
    if cases != 0:
        print()



